sql/core/src/test/scala/org/apache/spark/sql/execution/SparkSqlParserSuite.scala

```diff
@@ -371,4 +371,42 @@ class SparkSqlParserSuite extends AnalysisTest {
           ("TOK_TABLEROWFORMATNULL", "NULL")), None, None,
         List.empty, List.empty, None, None, false)))
   }
+  
+  test("SPARK-32607: Script Transformation ROW FORMAT DELIMITED" +
+    " `TOK_TABLEROWFORMATLINES` only support '\\n'") {
+
+      // test input format TOK_TABLEROWFORMATLINES
+      intercept(
+          s"""
+             |SELECT TRANSFORM(a, b, c, d, e)
+             |  ROW FORMAT DELIMITED
+             |  FIELDS TERMINATED BY ','
+             |  LINES TERMINATED BY '@'
+             |  NULL DEFINED AS 'null'
+             |  USING 'cat' AS (value)
+             |  ROW FORMAT DELIMITED
+             |  FIELDS TERMINATED BY '&'
+             |  LINES TERMINATED BY '\n'
```

Comment:
why does this fail? It's '\n' right?

Reply:
ah there are two `LINES TERMINATED BY`. Is it allowed?

Reply:
> why does this fail? It's '\n' right?

Scrip in/out, two case.